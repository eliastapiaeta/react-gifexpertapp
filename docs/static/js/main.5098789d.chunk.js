(this["webpackJsonpgif-expert-app"]=this["webpackJsonpgif-expert-app"]||[]).push([[0],{17:function(t,e,n){},19:function(t,e,n){"use strict";n.r(e);var a=n(0),r=n(1),i=n(7),c=n.n(i),s=n(2),u=n(9),o=function(t){var e=t.setCategorias,n=Object(r.useState)(""),i=Object(s.a)(n,2),c=i[0],o=i[1],d=Object(r.useState)(10),j=Object(s.a)(d,2),l=j[0],b=j[1],f=function(t,e){var n=null!==t&&t,a="string"!==typeof n?n.toString():n,r=a?a.match(/(\d+)/g):null,i=null!==r&&r,c=i&&i.toString().substring(0,e);return Number(c)},m=function(t){for(var e="",n=0;n<t.length;n++)-1!==" abcdefghijklmn\xf1opqrstuvwxyzABCDEFGHIJKLMN\xd1OPQRSTUVWXYZ1234567890".indexOf(t.charAt(n))&&(e+=t.charAt(n));return e},p=function(t,e){var n=t.length;return{valorLimitado:n>=0&&n<=e&&!0,valorAlcanzado:e===n&&!0}};return Object(a.jsx)("form",{onSubmit:function(t){return function(t){t.preventDefault(),c.trim().length>2&&""!==l&&l&&(e((function(t){return[{num:l,cat:c}].concat(Object(u.a)(t))})),o(""))}(t)},children:Object(a.jsxs)("div",{id:"formGif",children:[Object(a.jsx)("div",{id:"textGifs",children:Object(a.jsxs)("label",{children:["Imagen",Object(a.jsx)("input",{id:"inText",placeholder:"Imagen a buscar",type:"text",maxLength:"25",pattern:"[A-Za-z0-9 ]{1,30}",value:c,onChange:function(t){var e=m(t.target.value);p(e,25)&&o(e)}})]})}),Object(a.jsx)("div",{id:"numberGifs",children:Object(a.jsxs)("label",{children:["Lote",Object(a.jsx)("input",{id:"inNum",placeholder:"n\xb0",type:"text",maxLength:"3",pattern:"[0-9]{1,2}",value:l,onChange:function(t){var e=""!==t.target.value&&t.target.value,n=f(e,2);(n&&p(n.toString(),2)).valorLimitado&&b(n),n||b("")}})]})}),Object(a.jsx)("div",{id:"divEnter",children:Object(a.jsxs)("label",{children:[Object(a.jsx)("br",{}),Object(a.jsx)("input",{type:"submit",id:"btnForm",value:"Ok"})]})}),Object(a.jsx)("br",{id:"brGifs"})]})})},d=n(10),j=n(6),l=n.n(j),b=n(8),f=function(){var t=Object(b.a)(l.a.mark((function t(e,n){var a,r,i,c,s;return l.a.wrap((function(t){for(;;)switch(t.prev=t.next){case 0:return a="https://api.giphy.com/v1/gifs/search?q=".concat(encodeURI(e),"&limit=").concat(encodeURI(n),"&rating=pg&api_key=Y9ZpxPqgrW8xKcg06GWFeWUFfQGGKTiX"),t.next=3,fetch(a);case 3:return r=t.sent,t.next=6,r.json();case 6:return i=t.sent,c=i.data,s=c.map((function(t){var e;return{id:t.id,title:t.title,url:null===(e=t.images)||void 0===e?void 0:e.downsized_medium.url}})),t.abrupt("return",s);case 10:case"end":return t.stop()}}),t)})));return function(e,n){return t.apply(this,arguments)}}(),m=function(t){var e=t.id,n=t.title,r=t.url;return Object(a.jsxs)("div",{className:"card animate__animated animate__fadeInLeft",children:[Object(a.jsx)("img",{id:e,src:r,alt:n}),Object(a.jsx)("p",{children:n})]})},p=function(t){var e=t.categoria,n=t.quantity,i=function(t,e){var n=Object(r.useState)({data:[],loading:!0}),a=Object(s.a)(n,2),i=a[0],c=a[1];return Object(r.useEffect)((function(){f(t,e).then((function(t){c({data:t,loading:!1})}))}),[t,e]),i}(e,n),c=i.data,u=i.loading;return Object(a.jsxs)(a.Fragment,{children:[Object(a.jsxs)("h3",{className:"animate__animated animate__bounceIn",children:[e," (",n,")"]}),u&&Object(a.jsx)("p",{className:"animate__animated animate__flash",children:"Loading..."}),Object(a.jsx)("div",{className:"card-grid",children:c.map((function(t){return Object(a.jsx)(m,Object(d.a)({},t),t.id)}))})]})},h=function(t){var e=t.defaultCategories,n=void 0===e?[]:e,i=Object(r.useState)(n),c=Object(s.a)(i,2),u=c[0],d=c[1];return Object(a.jsxs)(a.Fragment,{children:[Object(a.jsx)("h2",{children:"GifExpertApp"}),Object(a.jsx)(o,{setCategorias:d}),Object(a.jsx)("hr",{}),u.map((function(t,e){return Object(a.jsx)(p,{categoria:t.cat,quantity:t.num},t+e)}))]})};n(17),n(18);c.a.render(Object(a.jsx)(h,{}),document.getElementById("root"))}},[[19,1,2]]]);
//# sourceMappingURL=main.5098789d.chunk.js.map