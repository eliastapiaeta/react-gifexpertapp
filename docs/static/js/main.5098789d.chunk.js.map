{"version":3,"sources":["components/AddCategory.js","helpers/getGifs.js","components/GifGridItem.js","components/GifGrid.js","hooks/useFetchGifs.js","base/GifExpertApp.js","index.js"],"names":["AddCategory","setCategorias","useState","inputValue","setinputValue","inputNumberValue","setInputNumberValue","stringToNum","valNaN","cantidad","valNull","valString","toString","num","match","haveNumber","resulString","substring","Number","numtext","val","out","i","length","indexOf","charAt","limLengthStr","valor","limite","vmax","valorLimitado","valorAlcanzado","onSubmit","e","preventDefault","trim","cats","cat","handleSubmit","id","placeholder","type","maxLength","pattern","value","onChange","valorFiltrado","target","inValue","getGifs","categoria","quantity","a","url","encodeURI","fetch","resp","json","data","gifs","map","img","title","images","downsized_medium","GifGridItem","className","src","alt","GifGrid","loading","state","setstate","useEffect","then","imgs","useFetchGifs","GifExpertApp","defaultCategories","categorias","category","ReactDOM","render","document","getElementById"],"mappings":"mMAGaA,EAAc,SAAC,GAAuB,IAArBC,EAAoB,EAApBA,cAAoB,EAEVC,mBAAS,IAFC,mBAEvCC,EAFuC,KAE3BC,EAF2B,OAGEF,mBAAS,IAHX,mBAGvCG,EAHuC,KAGrBC,EAHqB,KAwDxCC,EAAc,SAACC,EAAQC,GAEzB,IAAMC,EAAsB,OAAXF,GAAoBA,EAM/BG,EAAgC,kBAAZD,EAAwBA,EAAQE,WAAaF,EAIjEG,EAAMF,EAAYA,EAAUG,MATpB,UASmC,KAC3CC,EAAsB,OAARF,GAAiBA,EAI/BG,EAAeD,GAAgBA,EAAWH,WAAYK,UAAU,EAAGR,GAGzE,OAFoBS,OAAOF,IAIzBG,EAAU,SAACC,GAKb,IAJA,IAAIC,EAAM,GAIDC,EAAI,EAAGA,EAAIF,EAAIG,OAAQD,KACW,IAH9B,0EAGEE,QAAQJ,EAAIK,OAAOH,MAC1BD,GAAOD,EAAIK,OAAOH,IAC1B,OAAOD,GAELK,EAAe,SAACC,EAAOC,GAEzB,IACMC,EAAOF,EAAMJ,OAKnB,MAAO,CAAEO,cADeD,GALX,GAK6BA,GAAQD,IAAY,EACtCG,eAJAH,IAAWC,IAAS,IAOhD,OACI,sBAAMG,SAAU,SAACC,GAAD,OAhEC,SAACA,GAClBA,EAAEC,iBAGE/B,EAAWgC,OAAOZ,OAAS,GACY,KAArBlB,GAA4BA,IAS1CJ,GAAc,SAAAmC,GAAI,OAAK,CAAEvB,IAAKR,EAAkBgC,IAAKlC,IAAnC,mBAAoDiC,OACtEhC,EAAc,KAiDCkC,CAAaL,IAApC,SACI,sBAAKM,GAAG,UAAR,UACI,qBAAKA,GAAG,WAAR,SACI,2CAEI,uBACIA,GAAG,SACHC,YAAY,kBACZC,KAAK,OACLC,UAAU,KACVC,QAAQ,qBACRC,MAAOzC,EACP0C,SA3GM,SAACZ,GAC3B,IAAMa,EAAgB3B,EAAQc,EAAEc,OAAOH,OACjBlB,EAAaoB,EAAe,KAG9C1C,EAAc0C,WA0GV,qBAAKP,GAAG,aAAR,SACI,yCAEI,uBACIA,GAAG,QACHC,YAAY,QACZC,KAAK,OACLC,UAAU,IACVC,QAAQ,aACRC,MAAOvC,EACPwC,SA/GK,SAACZ,GAC1B,IACMe,EAA8B,KAAnBf,EAAEc,OAAOH,OAAiBX,EAAEc,OAAOH,MAC9CE,EAAgBvC,EAAYyC,EAFjB,IAGUF,GAAkBpB,EAAaoB,EAAclC,WAHvD,IAGTkB,eASJxB,EAAoBwC,GAInBA,GACDxC,EAAoB,YAiGhB,qBAAKiC,GAAG,WAAR,SACI,kCACI,uBACA,uBAAOE,KAAK,SAASF,GAAG,UAAUK,MAAM,YAGhD,oBAAIL,GAAG,iB,+BC3IVU,EAAO,uCAAG,WAAOC,EAAWC,GAAlB,uBAAAC,EAAA,6DAIbC,EAJa,iDAImCC,UAAUJ,GAJ7C,kBAIiEI,UAAUH,GAJ3E,gEAKAI,MAAMF,GALN,cAKbG,EALa,gBAMIA,EAAKC,OANT,uBAMXC,EANW,EAMXA,KAEFC,EAAOD,EAAKE,KAAI,SAAAC,GAAQ,IAAD,EACzB,MAAO,CACHtB,GAAIsB,EAAItB,GACRuB,MAAOD,EAAIC,MACXT,IAAG,UAAEQ,EAAIE,cAAN,aAAE,EAAYC,iBAAiBX,QAZvB,kBAgBZM,GAhBY,4CAAH,wDCGPM,EAAc,SAAC,GAAwB,IAAtB1B,EAAqB,EAArBA,GAAIuB,EAAiB,EAAjBA,MAAOT,EAAU,EAAVA,IAErC,OACI,sBAAKa,UAAU,6CAAf,UACI,qBAAK3B,GAAIA,EAAI4B,IAAKd,EAAKe,IAAKN,IAC5B,4BAAIA,QCFHO,EAAU,SAAC,GAA6B,IAA3BnB,EAA0B,EAA1BA,UAAWC,EAAe,EAAfA,SAAe,ECHxB,SAACD,EAAWC,GAAc,IAAD,EAEvBjD,mBAAS,CAC/BwD,KAAM,GACNY,SAAS,IAJoC,mBAE1CC,EAF0C,KAEnCC,EAFmC,KA0BjD,OAnBAC,qBAAU,WAENxB,EAAQC,EAAWC,GACduB,MAAK,SAAAC,GACFH,EAAS,CACLd,KAAMiB,EACNL,SAAS,SAItB,CAACpB,EAAWC,IASRoB,EDpB2BK,CAAa1B,EAAWC,GAA5CY,EAHkC,EAGxCL,KAAcY,EAH0B,EAG1BA,QAEtB,OACI,qCACI,qBAAIJ,UAAU,sCAAd,UAAqDhB,EAArD,KAAkEC,EAAlE,OAGCmB,GAAW,mBAAGJ,UAAU,mCAAb,wBAEX,qBAAKA,UAAU,YAAf,SAEOH,EAAOH,KAAI,SAAAC,GAAG,OACV,cAAC,EAAD,eAEQA,GADCA,EAAItB,aE+BtBsC,EAjDM,SAAC,GAAgC,IAAD,IAA7BC,yBAA6B,MAAT,GAAS,IAYb5E,mBAAS4E,GAZI,mBAY1CC,EAZ0C,KAY9B9E,EAZ8B,KAcjD,OAAO,qCAEH,8CACA,cAAC,EAAD,CAAaA,cAAeA,IAC5B,uBAKI8E,EAAWnB,KAAI,SAACoB,EAAU1D,GAAX,OAKX,aAJA,CAIC,EAAD,CAEI4B,UAAW8B,EAAS3C,IACpBc,SAAU6B,EAASnE,KAFdmE,EAAW1D,U,YC1BpC2D,IAASC,OACP,cAAC,EAAD,IACAC,SAASC,eAAe,W","file":"static/js/main.5098789d.chunk.js","sourcesContent":["import React, { useState } from 'react'\r\nimport PropTypes from 'prop-types';\r\n\r\nexport const AddCategory = ({ setCategorias }) => {\r\n\r\n    const [inputValue, setinputValue] = useState('');\r\n    const [inputNumberValue, setInputNumberValue] = useState(10);\r\n\r\n    const handleInputChangeText = (e) => {\r\n        const valorFiltrado = numtext(e.target.value);\r\n        const valorLimitado = limLengthStr(valorFiltrado, 25);\r\n\r\n        if (valorLimitado) {\r\n            setinputValue(valorFiltrado);\r\n            // console.log('inputValue: ' + inputValue);\r\n            // console.log('valorFiltrado: ' + valorFiltrado);\r\n        }\r\n    }\r\n    const handleInputChangeNum = (e) => {\r\n        const cantidad = 2;\r\n        const inValue = (e.target.value !== \"\") && e.target.value;\r\n        const valorFiltrado = stringToNum(inValue, cantidad);\r\n        const { valorLimitado } = (valorFiltrado) && limLengthStr(valorFiltrado.toString(), cantidad);\r\n\r\n        // console.log('\\n');\r\n        // console.log('inValue: ' + inValue);\r\n        // console.log('valorFiltrado: ' + valorFiltrado);\r\n        // console.log('valorLimitado: ' + valorLimitado);\r\n        // console.log('valorAlcanzado: ' + valorAlcanzado);\r\n        // console.log('\\n');\r\n        if (valorLimitado) {\r\n            setInputNumberValue(valorFiltrado);\r\n            // console.log('inputValue: ' + inputValue);\r\n            // console.log('valorFiltrado: ' + valorFiltrado);\r\n        }\r\n        if (!valorFiltrado) {\r\n            setInputNumberValue('');\r\n        }\r\n    }\r\n    const handleSubmit = (e) => {\r\n        e.preventDefault();\r\n        // console.log('handleSubmit', inputValue);\r\n\r\n        if (inputValue.trim().length > 2) {\r\n            const inputNum = (inputNumberValue !== \"\") && inputNumberValue;\r\n\r\n            // console.log('inputNum: ' + inputNum);\r\n\r\n            if (inputNum) {\r\n                // console.log('submit OK');\r\n                // console.log('inputValue ', inputValue, 'typeof inputValue', typeof inputValue);\r\n                // console.log('inputNumberValue ', inputNumberValue, 'typeof inputNumberValue', typeof inputNumberValue);\r\n\r\n                setCategorias(cats => [{ num: inputNumberValue, cat: inputValue }, ...cats]);\r\n                setinputValue('');\r\n            }\r\n        }\r\n\r\n    }\r\n    const stringToNum = (valNaN, cantidad) => {\r\n\r\n        const valNull = (valNaN !== null) && valNaN;\r\n        const regex = /(\\d+)/g;\r\n\r\n        // console.log('valNaN: ' + valNaN + ' typeof: ' + typeof valNaN);\r\n        // console.log('valNull: ' + valNull + ' typeof: ' + typeof valNull);\r\n\r\n        const valString = (typeof valNull !== 'string') ? valNull.toString() : valNull;\r\n\r\n        // console.log('valString: ' + valString + ' typeof: ' + typeof valString);\r\n\r\n        const num = valString ? valString.match(regex) : null;\r\n        const haveNumber = (num !== null) && num;\r\n\r\n        // console.log('num:::: ' + typeof haveNumber + ' ' + haveNumber);\r\n\r\n        const resulString = (haveNumber) && (haveNumber.toString()).substring(0, cantidad);\r\n        const resulNumber = Number(resulString);\r\n\r\n        return resulNumber;\r\n    }\r\n    const numtext = (val) => {\r\n        let out = '';\r\n        //Se añaden las letras validas\r\n        let filtro = ' abcdefghijklmnñopqrstuvwxyzABCDEFGHIJKLMNÑOPQRSTUVWXYZ1234567890';//Caracteres validos\r\n\r\n        for (let i = 0; i < val.length; i++)\r\n            if (filtro.indexOf(val.charAt(i)) !== -1)\r\n                out += val.charAt(i);\r\n        return out;\r\n    }\r\n    const limLengthStr = (valor, limite) => {\r\n        // console.log(limite);\r\n        const vmin = 0;\r\n        const vmax = valor.length;\r\n        const valorAlcanzado = (limite === vmax) && true;\r\n        // console.log('\\n valor: ' + valor);\r\n        // console.log('vmin: ' + vmin + ' vmax: ' + vmax);\r\n        const valorLimitado = ((vmax >= vmin) && (vmax <= limite)) && true;\r\n        return { valorLimitado, valorAlcanzado };\r\n    }\r\n\r\n    return (\r\n        <form onSubmit={(e) => handleSubmit(e)}>\r\n            <div id='formGif'>\r\n                <div id='textGifs'>\r\n                    <label>\r\n                        Imagen\r\n                        <input\r\n                            id=\"inText\"\r\n                            placeholder='Imagen a buscar'\r\n                            type='text'\r\n                            maxLength='25'\r\n                            pattern=\"[A-Za-z0-9 ]{1,30}\"\r\n                            value={inputValue}\r\n                            onChange={handleInputChangeText}\r\n                        />\r\n                    </label>\r\n                </div>\r\n                <div id='numberGifs'>\r\n                    <label>\r\n                        Lote\r\n                        <input\r\n                            id='inNum'\r\n                            placeholder='n°'\r\n                            type='text'\r\n                            maxLength='3'\r\n                            pattern='[0-9]{1,2}'\r\n                            value={inputNumberValue}\r\n                            onChange={handleInputChangeNum}\r\n                        />\r\n                    </label>\r\n                </div>\r\n                <div id='divEnter'>\r\n                    <label>\r\n                        <br />\r\n                        <input type=\"submit\" id=\"btnForm\" value=\"Ok\" />\r\n                    </label>\r\n                </div>\r\n                <br id='brGifs' />\r\n            </div>\r\n        </form>\r\n    )\r\n}\r\n\r\nAddCategory.propTypes = {\r\n    setCategorias: PropTypes.func.isRequired\r\n}\r\n","export const getGifs = async (categoria, quantity) => {\r\n\r\n    // console.log(categoria, 'quantity: ' , quantity, ' typeof quantity ' , typeof quantity);\r\n    \r\n    const url = `https://api.giphy.com/v1/gifs/search?q=${encodeURI(categoria)}&limit=${encodeURI(quantity)}&rating=pg&api_key=Y9ZpxPqgrW8xKcg06GWFeWUFfQGGKTiX`;\r\n    const resp = await fetch(url);\r\n    const { data } = await resp.json();\r\n\r\n    const gifs = data.map(img => {\r\n        return {\r\n            id: img.id,\r\n            title: img.title,\r\n            url: img.images?.downsized_medium.url\r\n        }\r\n    })\r\n\r\n    return gifs;\r\n}","import React from 'react'\r\nimport PropTypes from 'prop-types';\r\n\r\nexport const GifGridItem = ({ id, title, url }) => {\r\n    // console.log(id, title, url);\r\n    return (\r\n        <div className='card animate__animated animate__fadeInLeft'>\r\n            <img id={id} src={url} alt={title} />\r\n            <p>{title}</p>\r\n        </div>\r\n    )\r\n}\r\n\r\nGifGridItem.propTypes = {\r\n    id : PropTypes.string.isRequired,\r\n    title : PropTypes.string.isRequired,\r\n    url: PropTypes.string.isRequired\r\n}\r\n\r\n/**\r\n * 1. Enzymme\r\n * 2. Enzyme to Json\r\n * 3. Debe mostrar el componente correctamente\r\n *      * shallow\r\n *      * wrapper\r\n *      * wrapper .toMatchSnapshot()\r\n */\r\n","import React, { /* useState, useEffect */ } from 'react';\r\nimport { useFetchGifs } from '../hooks/useFetchGifs';\r\nimport { GifGridItem } from './GifGridItem';\r\nimport PropTypes from 'prop-types';\r\n\r\n\r\nexport const GifGrid = ({ categoria, quantity }) => {\r\n\r\n    // console.log(categoria, quantity);\r\n    const { data: images, loading } = useFetchGifs(categoria, quantity);\r\n\r\n    return (\r\n        <>\r\n            <h3 className='animate__animated animate__bounceIn'>{categoria} ({quantity})</h3>\r\n\r\n            { /* loading ? 'Cargando...' : 'Data cargada' */}\r\n            {loading && <p className='animate__animated animate__flash'>Loading...</p>}\r\n\r\n            {<div className=\"card-grid\">\r\n                {\r\n                    images.map(img => (\r\n                        <GifGridItem\r\n                            key={img.id}\r\n                            {...img}\r\n                        />\r\n                        // <li key={id}>{title}</li>\r\n                    ))\r\n                } \r\n            </div>}\r\n        </>\r\n    )\r\n}\r\n\r\nGifGrid.propTypes = {\r\n    categoria: PropTypes.string.isRequired,\r\n    quantity: PropTypes.number.isRequired    \r\n}\r\n","import { useEffect, useState } from 'react';\r\nimport { getGifs } from '../helpers/getGifs';\r\n\r\nexport const useFetchGifs = (categoria, quantity) => {\r\n\r\n    const [state, setstate] = useState({\r\n        data: [],\r\n        loading: true\r\n    });\r\n\r\n    useEffect(() => {\r\n\r\n        getGifs(categoria, quantity)\r\n            .then(imgs => {\r\n                setstate({\r\n                    data: imgs,\r\n                    loading: false\r\n                })\r\n            })\r\n\r\n    }, [categoria, quantity])\r\n\r\n    /*   setTimeout(() => {\r\n          setstate({\r\n              data: [1, 2, 3, 4, 5, 6, 7],\r\n              loading: false\r\n          })\r\n      }, 3000); */\r\n\r\n    return state;  // { data:[], loading:true }\r\n}\r\n\r\n","import { useState } from \"react\";\r\nimport { AddCategory } from \"../components/AddCategory\";\r\nimport { GifGrid } from '../components/GifGrid.js';\r\n\r\nconst GifExpertApp = ({ defaultCategories = [] }) => {\r\n\r\n    // const categorias = ['One Punch', 'Samurai X', 'Dragon ball'];\r\n\r\n    // const [categorias, setCategorias] = useState(['One Punch', 'Samurai X', 'Dragon ball']);\r\n    /* \r\n        const handleAdd = () => {\r\n            // setCategorias([...categorias,'hunterXhunter']);\r\n            setCategorias(cats => [...cats, 'hunterXhunter']);\r\n        }\r\n     */\r\n\r\n    const [categorias, setCategorias] = useState(defaultCategories);\r\n\r\n    return <>\r\n\r\n        <h2>GifExpertApp</h2>\r\n        <AddCategory setCategorias={setCategorias} />\r\n        <hr />\r\n\r\n        {\r\n            // console.log(categorias)\r\n            // categorias === 'undefinded' && console.log(categorias[0])\r\n            categorias.map((category, i) =>\r\n                // <li>{category}</li>\r\n                // console.log(category)\r\n\r\n                // limpiar(category.num)\r\n                <GifGrid\r\n                    key={category + i}\r\n                    categoria={category.cat}\r\n                    quantity={category.num}\r\n\r\n                />\r\n                // category.map(cat => {\r\n                // let texto = ((typeof cat.text) !== 'undefined') ? cat.text : false;\r\n                // let num = ((typeof cat.num) !== 'undefined') ? cat.num : false;\r\n                // console.log(cat.text);\r\n                // })\r\n                // <GifGrid\r\n                //     key={category + i}\r\n                //     categoria={category}\r\n                // />\r\n            )\r\n        }\r\n    </>\r\n\r\n}\r\n\r\nexport default GifExpertApp;\r\n","import React from 'react';\nimport ReactDOM from 'react-dom';\nimport GifExpertApp from './base/GifExpertApp';\nimport './index.css';\nimport 'animate.css';\n\n\nReactDOM.render(\n  <GifExpertApp />,\n  document.getElementById('root')\n);\n\n// Tarea\n/**\n *  GifExpertApp\n *  <h2>GifExpertApp</h2>\n *  <hr />\n */\n"],"sourceRoot":""}