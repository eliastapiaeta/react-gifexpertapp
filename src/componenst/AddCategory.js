import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategorias }) => {

    const [inputValue, setinputValue] = useState('');
    const [inputNumberValue, setInputNumberValue] = useState(10);

    const handleInputChangeText = (e) => {
        setinputValue(e.target.value);
    }
    const handleInputChangeNum = (e) => {
        setInputNumberValue(Number(e.target.value));
    }
    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 2) {
            setCategorias(cats => [{ num: inputNumberValue, cat: inputValue }, ...cats]);
            setinputValue('');
        }

    }

    return (
        <form onSubmit={(e) => handleSubmit(e)}>
            <div id='formGif'>
                <div id='textGifs'>
                    <label>
                        Imagen
                        <input
                            placeholder='Imagen a buscar'
                            type='text'
                            value={inputValue}
                            onChange={handleInputChangeText}
                        // onKeyDown={(e) => e.key === 'Enter' && onsubmit}
                        />
                    </label>
                </div>
                <div id='numberGifs'>
                    <label>
                        Cantidad
                        <input
                            placeholder='Cantidad'
                            type='text'
                            size='11'
                            pattern='[0-9]{1,2}'
                            value={inputNumberValue}
                            onChange={handleInputChangeNum}
                        />
                    </label>
                </div>
                <input type="submit" id="submitbtn" />
                <br id='brGifs' />
            </div>
            <div>{inputValue} ({inputNumberValue})</div>
        </form>
    )
}

AddCategory.propTypes = {
    setCategorias: PropTypes.func.isRequired
}
