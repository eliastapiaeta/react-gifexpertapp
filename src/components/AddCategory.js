import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategorias }) => {

    const [inputValue, setinputValue] = useState('');
    const [inputNumberValue, setInputNumberValue] = useState(10);

    const handleInputChangeText = (e) => {
        const valorFiltrado = numtext(e.target.value);
        const valorLimitado = limLengthStr(valorFiltrado, 25);

        if (valorLimitado) {
            setinputValue(valorFiltrado);
            // console.log('inputValue: ' + inputValue);
            // console.log('valorFiltrado: ' + valorFiltrado);
        }
    }
    const handleInputChangeNum = (e) => {
        const cantidad = 2;
        const inValue = (e.target.value !== "") && e.target.value;
        const valorFiltrado = stringToNum(inValue, cantidad);
        const { valorLimitado } = (valorFiltrado) && limLengthStr(valorFiltrado.toString(), cantidad);

        // console.log('\n');
        // console.log('inValue: ' + inValue);
        // console.log('valorFiltrado: ' + valorFiltrado);
        // console.log('valorLimitado: ' + valorLimitado);
        // console.log('valorAlcanzado: ' + valorAlcanzado);
        // console.log('\n');
        if (valorLimitado) {
            setInputNumberValue(valorFiltrado);
            // console.log('inputValue: ' + inputValue);
            // console.log('valorFiltrado: ' + valorFiltrado);
        }
        if (!valorFiltrado) {
            setInputNumberValue('');
        }
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        // console.log('handleSubmit', inputValue);

        if (inputValue.trim().length > 2) {
            const inputNum = (inputNumberValue !== "") && inputNumberValue;

            // console.log('inputNum: ' + inputNum);

            if (inputNum) {
                // console.log('submit OK');
                // console.log('inputValue ', inputValue, 'typeof inputValue', typeof inputValue);
                // console.log('inputNumberValue ', inputNumberValue, 'typeof inputNumberValue', typeof inputNumberValue);

                setCategorias(cats => [{ num: inputNumberValue, cat: inputValue }, ...cats]);
                setinputValue('');
            }
        }

    }
    const stringToNum = (valNaN, cantidad) => {

        const valNull = (valNaN !== null) && valNaN;
        const regex = /(\d+)/g;

        // console.log('valNaN: ' + valNaN + ' typeof: ' + typeof valNaN);
        // console.log('valNull: ' + valNull + ' typeof: ' + typeof valNull);

        const valString = (typeof valNull !== 'string') ? valNull.toString() : valNull;

        // console.log('valString: ' + valString + ' typeof: ' + typeof valString);

        const num = valString ? valString.match(regex) : null;
        const haveNumber = (num !== null) && num;

        // console.log('num:::: ' + typeof haveNumber + ' ' + haveNumber);

        const resulString = (haveNumber) && (haveNumber.toString()).substring(0, cantidad);
        const resulNumber = Number(resulString);

        return resulNumber;
    }
    const numtext = (val) => {
        let out = '';
        //Se añaden las letras validas
        let filtro = ' abcdefghijklmnñopqrstuvwxyzABCDEFGHIJKLMNÑOPQRSTUVWXYZ1234567890';//Caracteres validos

        for (let i = 0; i < val.length; i++)
            if (filtro.indexOf(val.charAt(i)) !== -1)
                out += val.charAt(i);
        return out;
    }
    const limLengthStr = (valor, limite) => {
        // console.log(limite);
        const vmin = 0;
        const vmax = valor.length;
        const valorAlcanzado = (limite === vmax) && true;
        // console.log('\n valor: ' + valor);
        // console.log('vmin: ' + vmin + ' vmax: ' + vmax);
        const valorLimitado = ((vmax >= vmin) && (vmax <= limite)) && true;
        return { valorLimitado, valorAlcanzado };
    }

    return (
        <form onSubmit={(e) => handleSubmit(e)}>
            <div id='formGif'>
                <div id='textGifs'>
                    <label>
                        Imagen
                        <input
                            id="inText"
                            placeholder='Imagen a buscar'
                            type='text'
                            maxLength='25'
                            pattern="[A-Za-z0-9 ]{1,30}"
                            value={inputValue}
                            onChange={handleInputChangeText}
                        />
                    </label>
                </div>
                <div id='numberGifs'>
                    <label>
                        Lote
                        <input
                            id='inNum'
                            placeholder='n°'
                            type='text'
                            maxLength='3'
                            pattern='[0-9]{1,2}'
                            value={inputNumberValue}
                            onChange={handleInputChangeNum}
                        />
                    </label>
                </div>
                <div id='divEnter'>
                    <label>
                        <br />
                        <input type="submit" id="btnForm" value="Ok" />
                    </label>
                </div>
                <br id='brGifs' />
            </div>
        </form>
    )
}

AddCategory.propTypes = {
    setCategorias: PropTypes.func.isRequired
}
