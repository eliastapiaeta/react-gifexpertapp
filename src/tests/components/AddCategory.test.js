import { AddCategory } from "../../components/AddCategory.js";
const { shallow } = require("enzyme");

describe('Pruebas en el componente <AddCategory />', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategorias={setCategories} />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategorias={setCategories} />);
    });

    test('should mostrarse correctamente', () => {

        expect(wrapper).toMatchSnapshot();
    });
    test('should cambiar la caja de texto busqueda', () => {
        const value = 'Hola Mundo';
        const input = wrapper.find('#inText');
        input.simulate('change', { target: { value } });

        const returnedValue = wrapper.find('#inText').props().value.trim();
        expect(returnedValue).toBe(value);
    });
    test('no debe de postear la informaciÃ³n con submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault() { } });

        expect(setCategories).not.toHaveBeenCalled();
    });
    test('should llamar el setCategories y limpiar la caja de texto', () => {
        // 1. Simular el imputhange
        // 2. Simular el submit
        // 3. setCategories se debe de haber llamado
        // 4. El valor del input debe de estar ''

        const valueText = 'faithfulness';
        const valueNum = '3';

        // 1. Simular el imputhange
        wrapper.find('#inText').simulate('change', { target: { value: valueText } });
        wrapper.find('#inNum').simulate('change', { target: { value: valueNum } });
        // 2. Simular el submit
        wrapper.find('form').simulate('submit', { preventDefault() { } });

        const valTextFromForm = wrapper.find('#inText').props().value;

        // 3. setCategories se debe de haber llamado
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        // 4. El valor del input debe de estar ''
        expect(valTextFromForm).toBe('');
    });

})
