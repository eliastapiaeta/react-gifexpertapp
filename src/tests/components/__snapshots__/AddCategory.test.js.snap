// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <AddCategory /> should mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    id="formGif"
  >
    <div
      id="textGifs"
    >
      <label>
        Imagen
        <input
          id="inText"
          maxLength="25"
          onChange={[Function]}
          pattern="[A-Za-z0-9 ]{1,30}"
          placeholder="Imagen a buscar"
          type="text"
          value=""
        />
      </label>
    </div>
    <div
      id="numberGifs"
    >
      <label>
        Lote
        <input
          id="inNum"
          maxLength="3"
          onChange={[Function]}
          pattern="[0-9]{1,2}"
          placeholder="nÂ°"
          type="text"
          value={10}
        />
      </label>
    </div>
    <div
      id="divEnter"
    >
      <label>
        <br />
        <input
          id="btnForm"
          type="submit"
          value="Ok"
        />
      </label>
    </div>
    <br
      id="brGifs"
    />
  </div>
</form>
`;
